lib/internal/Magento/Framework/ObjectManager/ObjectManager.php

```diff
@@ -53,7 +53,6 @@ public function __construct(FactoryInterface $factory, ConfigInterface $config,
      */
     public function create($type, array $arguments = [])
     {
-        $type = ltrim($type, '\\');
         return $this->_factory->create($this->_config->getPreference($type), $arguments);
```

Comment:
This does not look like a part of in-object cache, which is the purpose of this PR